### PR TITLE
Implementing Internationalization with SvelteKit and Directus

### PR DESCRIPTION
**Pull request by icode247**

### CODE REVIEW

File: multilingual-news-app-using-sveltekit-with-directus/index.md [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+Svelte is a new JavaScript framework for building user interfaces for web applications. It offer a better way for handling UI updates when compared with React, Angular, or Vue.js. In this tutorial, you'll learn how to build a multilingual news application using Sveltekit with Directus i18n API.
+
+## Before You Start
+You will need:
+- [Node.js v20.11.1](https://nodejs.org/) or later
+- A code editor on your computer.
+- Create and login to your [Directus Cloud account](https://directus.cloud/) and have install the Directus SDK `npm install @directus/sdk`.
+- Some knowledge of React and Svelte.
+
+The code for this tutorial is available on my [GitHub repository](https://github.com/icode247/directus-i18n-app); feel free to clone it to follow along.
```
Comment (phazonoverload) [inline multilingual-news-app-using-sveltekit-with-directus/index.md:19]:
```suggestion
## Before You Start

You will need:

- [Node.js v20.11.1](https://nodejs.org/) or later.
- A code editor on your computer.
- A Directus project - follow our [quickstart guide](https://docs.directus.io/getting-started/quickstart) if you don't already have one.
- Some knowledge of React and Svelte.

The code for this tutorial is available on my [GitHub repository](https://github.com/icode247/directus-i18n-app).
```

File: multilingual-news-app-using-sveltekit-with-directus/index.md [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+i18n is the process of creating applications that support multiple languages. The phrase i18n represents the first and the last letters of the word **internationalization**. The number 18 denotes the number of letters between the first and the last letter. The main goal of i18n is to conveniently create applications that support multiple languages or locations without having to redesign the application to add those languages.
+
+
+## Understanding Directus i18n API
+
+Directus is an open-source headless Content Management System that provides a robust API for managing content. It offers an i18n API that enables the internationalization and localization of content within the system.
+
+The Directus i18n API allows you to manage translations for various content types, such as articles, products, or any other data stored in the Directus system.
+
+With the Directus i18n API, you can build multilingual applications or websites that serve localized content to users based on their preferred language or region.
```
Comment (phazonoverload) [inline multilingual-news-app-using-sveltekit-with-directus/index.md:32]: Remove this - I don't think it adds much that hasn't already been said.

File: multilingual-news-app-using-sveltekit-with-directus/index.md [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+Next, change directory into the project folder and install the dependencies.
+
+```
+cd directus-i18n-app && npm install
+```
+
+Then install the **svelte-routing** package as a dev dependency:
+```
+npm i -D svelte-routing
+```
```
Comment (phazonoverload) [inline multilingual-news-app-using-sveltekit-with-directus/index.md:53]:
````suggestion
Let's start by setting up a new Svelte project using the [degit scaffolding tool](https://github.com/Rich-Harris/degit):

```
npx degit sveltejs/template directus-i18n-app
```

Install the required dependencies:

```
cd directus-i18n-app 
npm install
npm install @directus/sdk
npm install --save-dev svelte-routing
```

In the `src/libs` directory, create a `directus.js` file to create and export a Directus SDK instance:

```
import { createDirectus, rest } from '@directus/sdk';
const directus = createDirectus('<YOUR DIRECTUS URL>').with(rest());
export default directus;
```

Be sure to provide your real project URL.
````

This should be a continuation of 'Before You Start'. As you can see, we've brought all of the dependency installation into a single instruction to make this more concise. The SDK install has also been moved here as it's a related step.

I've also added the boring boilerplate of setting up the SDK to this section.

File: multilingual-news-app-using-sveltekit-with-directus/index.md [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+Then create some entries in your News collection.
+
+![Creating new entries in the news collections](./news-translation-entries.png)
+
+
+Now you can write your contents in different translations (German, French and English).
+
+Lastly allow the Public role to read the `news`, `languages` and `news_translations` collections in the Access Control settings.
+
+
```
Comment (phazonoverload) [inline multilingual-news-app-using-sveltekit-with-directus/index.md:110]:
```suggestion
## Designing a Data Model

In the Directus Data Studio, navigate to Settings -> Data Model and create a new collection called `news`:

- `id` (Primary Key Field, Type: Auto-incremented Integer)
- `author` (Type: String, Interface: Input)
- `cover` (Type: Image)

Create a collection called `languages`:

- `code` (Primary Key Field, Type: Manually entered string )
- `name` (Type: String, Interface: Input)
- `direction` (Type: String, Interface: Dropdown, Options: `ltr` and `rtl`. Default Value: `ltr`)

The `direction` field enables support for languages that read right to left.

To enable content translation in your `news` collection, add a new One to Many `translations` field with the `languages` collection. Select `name` as the **Language Indicator Field** and `direction` as the **Language Direction Field**.

![Creating Directus translations collection](./creating-news-translation-collection.png)

Once you save, a new collection named `news_translations` will be created for you. In the `news_translations` collection, you will add the fields that need translations. 

Add the following fields to the `news_translations` collection:

- `title` (Type: String, Interface: Input)
- `slug` (Type: String, Interface: Input)
- `body` (Type: Text, Interface: WYSIWYG)

Add each language you want to support as items in the `languages` collection. 

![Creating new entries in the languages collections](./languages-entries.png)

The item page for the `news` collection now includes a translations interface.

![Creating new entries in the news collections](./news-translation-entries.png)

Allow the Public role to read the `news`, `languages` and `news_translations` collections in the Access Control settings to ensure the frontend can access these fields.
```

I've made some suggestions to make this much more succinct, but I want to point out that you're shifting from second person (you) to first person plural (our) pronouns. This is good to keep an eye on as you write more content - stay consistent. I've changed them to second person for you.

File: multilingual-news-app-using-sveltekit-with-directus/index.md [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+## Building the News App Frontend with SvelteKit
+In your Svelte project, create a `directus.js` file to initialize Dirctus SDK in your project:
+
+```
+import { createDirectus, rest } from '@directus/sdk';
+
+const directus = createDirectus('<YOUR DIRECTUS URL>').with(rest());
+
+export default directus;
+```
```
Comment (phazonoverload) [inline multilingual-news-app-using-sveltekit-with-directus/index.md:120]: Remove

**Review by phazonoverload — CHANGES_REQUESTED** [review]
I've reviewed everything up until the part where you start on the Svelte application. Could you make sure you're following al lthe best practice from [this post](https://docs.directus.io/blog/getting-started-directus-sveltekit.html#create-a-wrapper-for-the-sdk) especially the bit I linked about creating a wrapper.

Could you please also remove all CSS from this project as it's not helping teach the core concepts.

Once you address the changes in the first half, and bring this more in like with the other SvelteKit post, request a re-review and I'll get right back on it. Thanks for your patience while I get around to providing this initial review.

**Comment by icode247** [comment]
Hi @phazonoverload,
The `readItem` function throws:

```
 errors: [
    {
      message: "You don't have permission to access this.",
      extensions: [Object]
    }
  ],
```

But I have given read permission to the collections. Also to works with the `readItems` function.

**Comment by icode247** [comment]
Hi @phazonoverload,
I have worked on the requested changes. Please take another look.

**Review by phazonoverload — APPROVED** [review]
Thank you for writing this, and we are happy to consider it done in it's current state. My team will schedule it in to be published. 

This repo is just for authoring/reviewing - the actual piece will be published via a Directus project, so we'll take it from here. I'll close this issue to keep the main branch clean, but this is a acceptance of your work.

Could you email me at [devrel@directus.io](mailto:devrel@directus.io) and we'll share what we need to get you paid. Thanks!

**Pull request closed without merging**